Source/Core/DiscIO/Volume.cpp

```diff
@@ -76,4 +82,49 @@ std::map<Language, std::string> IVolume::ReadWiiNames(const std::vector<u8>& dat
   }
   return names;
 }
+
+std::unique_ptr<IVolume> CreateVolumeFromFilename(const std::string& filename)
+{
+  std::unique_ptr<IBlobReader> reader(CreateBlobReader(filename));
+  if (reader == nullptr)
+    return nullptr;
+  CBlobBigEndianReader be_reader(*reader);
+
+  // Check for Wii
+  u32 wii_magic = 0;
+  be_reader.ReadSwapped(0x18, &wii_magic);
+  u32 wii_container_magic = 0;
+  be_reader.ReadSwapped(0x60, &wii_container_magic);
+  if (wii_magic == 0x5D1C9EA3 && wii_container_magic != 0)
+    return std::make_unique<CVolumeGC>(std::move(reader));
```

Comment:
To be honest, I can't write a good comment for this as I'm not completely sure how this is supposed to work. You can read more about it in PR #2047 if you want to. (And yes, the two year old comment at the end of that PR is still accurate. I'm planning to make a PR for it in a few weeks or so.)